kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -148,153 +151,116 @@ class ParallelDistanceCalculator
      */
-    void CalculateDistancesLagrangianSurface(ModelPart& rModelPart,
-            const Variable<double>& rDistanceVar,
-            const Variable<double>& rAreaVar,
-            const unsigned int max_levels,
-            const double max_distance)
+    void CalculateDistancesLagrangianSurface(
+        ModelPart& rModelPart,
+        const Variable<double>& rDistanceVar,
+        const Variable<double>& rAreaVar,
+        const unsigned int MaxLevels,
+        const double MaxDistance)
     {
         KRATOS_TRY
         bool is_distributed = false;
         if(rModelPart.GetCommunicator().TotalProcesses() > 1)
             is_distributed = true;
 
-        array_1d<double,TDim+1> visited;
-        const int elem_size = rModelPart.Elements().size();
-        const int node_size = rModelPart.Nodes().size();
-
         //check that variables needed are in the model part
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"distance Variable is not in the model part","");
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"Area Variable is not in the model part","");
-
+        const int node_size = rModelPart.Nodes().size();
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar))) << "Distance variable is not in the model part" << std::endl;
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar))) << "Area Variable is not in the model part" << std::endl;
         if(is_distributed == true)
-            if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX)) )
-                KRATOS_THROW_ERROR(std::logic_error,"PARTITION_INDEX Variable is not in the model part","");
+            KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX))) << "PARTITION_INDEX variable is not in the model part" << std::endl;
 
         // set to zero the distance
-        #pragma omp parallel for
-        for(int i = 0; i<node_size; i++)
-        {
-            ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
-            double& area = it->FastGetSolutionStepValue(rAreaVar);
-            area = 0.0;
-            double& is_visited = it->GetValue(IS_VISITED);
-            double& distance = it->FastGetSolutionStepValue(rDistanceVar);
-            it->GetValue(rDistanceVar) = it->FastGetSolutionStepValue(rDistanceVar);
-            if(is_visited != 1.0)
-            {
-                distance = 0.0;
+        block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+            double& r_area = rNode.FastGetSolutionStepValue(rAreaVar);
+            double& r_distance = rNode.FastGetSolutionStepValue(rDistanceVar);
+            rNode.GetValue(rDistanceVar) = r_distance;
+            if (rNode.Is(VISITED)) {
```

Comment:
@rubenzorrilla thank you for cleaning and updating this utility.

As I understand, IS_VISITED == 1.0 now becomes Is(VISITED). This condition was previously if(is_visited != 1.0). Please check it again.

Reply:
Yes. You're totally right. Thanks for noticing. In any case I think this function is no longer used (I'm planning to continue the clean up in future PRs).

Reply:
Done.